PDFCLOUD-2821 Add upload + raw JSON Python samples

datalogics-tsmith opened the pull request:
The intent of these samples is to live alongside the multi-part samples as an alternative way to make the same requests

datalogics-tsmith committed:
Add upload + raw JSON Python samples

datalogics-tsmith committed:
Added better file delineation to multi-file requests

Rather than file1 and file2 we refer to them as data/image/first/second/pdf
so that the code is hopefully more self documenting

datalogics-tsmith committed:
Change 'content-filename' to 'Content-Filename'

datalogics-tsmith committed:
Move new samples under "Endpoint Examples"

datalogics-tsmith committed:
Add READMEs to all of the "Endpoint Example" directories

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/html.py:
I get this error when I try to run this code:
```
HTML response status code: 501
{ "error":"Invalid endpoint. Please refer to documentation at https://pdfrest.com/documentation/" }
```

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/pdf-info.py:
```diff
@@ -0,0 +1,39 @@
+
+    uploaded_id = upload_response_json['files'][0]['id']
+    info_data = { "id" : uploaded_id, "queries": "title" }
+    print(json.dumps(info_data, indent = 2))Content-Filename
```
`Content-Filename` here causes a syntax error. Deleting it fixes it.

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/pdf-with-imported-form-data.py:
The final request in this sample gets sent to `/pdf-with-added-image` instead of `/pdf-with-imported-form-data`.

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/zip.py:
```diff
@@ -0,0 +1,55 @@
+print("Uploading second file...")
+upload_second_file_response = requests.post(url='https://api.pdfrest.com/upload',
+                    data=upload_second_file_data,
+                    headers={'Content-Type': 'application/octet-stream', 'content-filename': 'second_file.png', "API-Key": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"})
```
When I tested this I used two PDFs and ended up with two files: `first_file.pdf` and `second_file.png` but with the `second_file.png` actually being a PDF file with a `.png` extension. Should a comment be added that we assume that the first file is a PDF and the second is a PNG?

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/README.md:
```diff
@@ -0,0 +1,10 @@
+
+Using Python3:
+
+1. `cd Python/`
```
`cd 'Python/Endpoint Examples/JSON Payload/'` works for me here

datalogics-dliang reviewed (changes requested):
There are a few files that throw errors, but other than those the rest of the samples are working as expected.

datalogics-tsmith committed:
Minor fixes

- The /html route does not actually exist in production
- Extra words got copied/moved into the pdf-info sample
- Corrected imported data to call the right route
- Clarify naming for zip since it can take literally anything

datalogics-tsmith committed:
Correct Python example README paths for the new structure

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/README.md:
```diff
@@ -0,0 +1,10 @@
+
+Using Python3:
+
+1. `cd 'Python/Endpoint Examples/JSON Payload/`
```
This is missing the closing apostrophe, same thing in the Multipart Payload readme
```suggestion
1. `cd 'Python/Endpoint Examples/JSON Payload/'`
```

datalogics-dliang commented on Python/Endpoint Examples/Multipart Payload/README.md:
```diff
@@ -6,5 +6,5 @@ for accepted inputs, options, and other specifics about individual endpoints.
 Using Python3:
 
-1. `cd Python/`
+1. `cd 'Python/Endpoint Examples/Multipart Payload/`
```
This is missing the closing apostrophe
```suggestion
1. `cd 'Python/Endpoint Examples/Multipart Payload/'`
```

datalogics-dliang commented on Python/Endpoint Examples/JSON Payload/pdf-with-imported-form-data.py:
```diff
@@ -0,0 +1,55 @@
+
+    uploaded_pdf_id = upload_pdf_response_json['files'][0]['id']
+    uploaded_data_id = upload_data_response_json['files'][0]['id']
+    added_image_data = { "id" : uploaded_pdf_id, "data__file_id": uploaded_data_id }
```
`data__file_id` has an extra underscore
```suggestion
    added_image_data = { "id" : uploaded_pdf_id, "data_file_id": uploaded_data_id }
```

datalogics-dliang reviewed (changes requested):
Very minor changes left.

datalogics-tsmith committed:
More small tweaks
